TESTING.md

```diff
@@ -23,7 +23,7 @@ To run tests with go use these commands:
 
 ```bash
 cd govcd
-go test -check.v .
+go test -tags "functional" -check.v .
```

Comment:
It feels better to write up how to run all the tests here, because a newcomer may ignore some tests if he only runs functional.

Reply:
As mentioned above, `functional` means "all the tests using the gocheck library".
One of the reason for this set of changes is that we want to run tests on different frameworks without fear of clashes. For now we don't have such tests, but it will be soon. When we do, we will add instructions about them.

Reply:
@Didainius I believe you have some non-functional tests cooking? Would they work if executed together with the `functional` ones?

Reply:
OK, we agreed that either @dataclouder or @Didainius will add these new tests to the the `make test` once they are ready.